updatehub-sdk/examples/listener.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2018, 2019, 2020 O.S. Systems Sofware LTDA
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use updatehub_sdk::{listener, Result};
+
+#[async_std::main]
+async fn main() -> Result<()> {
+    let mut listener = listener::StateChangeListener::new();
+    listener.on_state_change(
+        listener::StateID::Download,
+        Box::new(|mut state| {
+            if state.id == listener::StateID::Download {
+                println!("Downloading");
+                state.cancel();
+                println!("Done");
+            }
+        }),
+    );
+
+    listener.on_state_change(
+        listener::StateID::Error,
+        Box::new(|state| {
+            if state.id == listener::StateID::Error {
+                println!("Error");
+            }
+        }),
+    );
+
+    listener.on_state_change(
+        listener::StateID::Error,
+        Box::new(|state| {
+            if state.id == listener::StateID::Reboot {
+                println!("Rebooting...");
+            }
+        }),
+    );
+
+    listener.on_error(Box::new(|_| {}));
```

Comment:
This handles doesn't follow the pattern used in other SDK, please take a look ait the [python's SDK example](https://github.com/UpdateHub/agent-sdk-python/blob/6d827d04478fd369f93d75904ca7e4dc4cb38d9e/examples/state_change_listener.py#L21-L47)